### PR TITLE
Set FCToken on right-click [GAWB-2198, prev. GAWB-2025]

### PR DESCRIPTION
**Pull request by aednichols**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/common/gcs_file_preview.cljs [hunk 1]
```diff
@@ -55,6 +55,7 @@
                          [:span {:style {:marginLeft "1em"}}
                           [:a {:href (common/gcs-object->download-url (:bucket-name props) (:object props))
                                :onClick #(utils/set-access-token-cookie (utils/get-access-token))
+                               :onContextMenu #(utils/set-access-token-cookie (utils/get-access-token))
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/common/gcs_file_preview.cljs]: You might `let` this function so you don't have to write it twice.

Comment (aednichols) [reply src/cljs/main/broadfcui/common/gcs_file_preview.cljs]: Like this?
```cljs
(let [refresh-token #(utils/set-access-token-cookie (utils/get-access-token))]
:onClick refresh-token
:onContextMenu refresh-token)
```
Is there a way to do it that keeps the function anonymous?

Comment (zarsky-broad) [reply src/cljs/main/broadfcui/common/gcs_file_preview.cljs]: Not without weird contortions.

Comment (MattPutnam) [reply src/cljs/main/broadfcui/common/gcs_file_preview.cljs]: Yeah, I would definitely prefer deduping that bit of code. Maybe it should be a util function?
```clojure
(defn refresh-access-token [] (set-access-token-cookie (get-access-token)))
```
in utils.cljs, then just `:onClick utils/refresh-access-token :onContextMenu utils/refresh-access-token` here.

Comment (aednichols) [reply src/cljs/main/broadfcui/common/gcs_file_preview.cljs]: Agreed, done.

File: src/cljs/main/broadfcui/common/gcs_file_preview.cljs [hunk 2]
```diff
@@ -54,7 +54,8 @@
                         (react/create-element
                          [:span {:style {:marginLeft "1em"}}
                           [:a {:href (common/gcs-object->download-url (:bucket-name props) (:object props))
-                               :onClick #(utils/set-access-token-cookie (utils/get-access-token))
+                               :onClick #(utils/refresh-access-token)
+                               :onContextMenu #(utils/refresh-access-token)
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/gcs_file_preview.cljs]: `#(foo)` can just be `foo`

Comment (aednichols) [reply src/cljs/main/broadfcui/common/gcs_file_preview.cljs]: Thanks, fixed

**Comment by MattPutnam** [comment]
👍

**Pull request merged**